ambry-store/src/main/java/com.github.ambry.store/BlobStoreStats.java

```diff
@@ -0,0 +1,567 @@
+package com.github.ambry.store;
+
+import com.github.ambry.utils.Pair;
+import com.github.ambry.utils.Time;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Exposes stats related to {@link BlobStore} that is useful to different components.
+ *
+ * Note: This is the v1 implementation of BlobStoreStats. The v1 implementation walks through the
+ * entire index and collect data needed to serve stats related requests for a predefined amount of
+ * time. Requests that are outside of this time range will trigger a new scan.
+ */
+public class BlobStoreStats implements StoreStats {
+  private final long bucketCount;
+  private final long bucketTimeSpan;
+  private final long segmentScanOffset;
+  private final boolean autoReschedule;
```

Comment:
BlobStoreStats will perform one scan on start (i.e. when start() is called) and not schedule for any scans at the end of the coverage period. The BlobStoreStats will serve stats requests until there is a request that asks for stats outside of the coverage period and trigger a new scan. The triggered scan will then update/extend the coverage period.